mantle/kola/tests/misc/network.go

```diff
@@ -84,30 +94,33 @@ NextProcess:
 			c.Fatalf("unexpected number of parts on line: %q in output %q", line, output)
 		}
 		proto := parts[0]
-		portdata := strings.Split(parts[3], ":")
-		port := portdata[len(portdata)-1]
-		pidProgramParts := strings.SplitN(parts[len(parts)-1], "/", 2)
-		if len(pidProgramParts) != 2 {
-			c.Errorf("%v did not contain pid and program parts; full output: %q", parts[6], output)
+		portData := strings.Split(parts[4], ":")
+		port := portData[len(portData)-1]
+		processData := parts[len(parts)-1]
+		pidStr := regexp.MustCompile(`pid=\d+`).FindString(processData)
+		processStr := regexp.MustCompile(`".+"`).FindString(processData)
+		if pidStr == "" || processStr == "" {
+			c.Errorf("%v did not contain pid and program; full output: %q", processData, output)
 			continue
 		}
-		pid, process := pidProgramParts[0], pidProgramParts[1]
-
-		for _, expected := range expectedListeners {
-			if strings.HasPrefix(proto, expected.protocol) && // allow expected tcp to match tcp6
-				expected.port == port &&
-				expected.process == process {
-				// matches expected process
-				continue NextProcess
-			}
+		pid, process := pidStr[4:], processStr[1:len(processStr)-1]
+
+		actual := listener{
+			process:  process,
+			protocol: proto,
+			port:     port,
+		}
+		if expectedListenersMap[actual] {
+			// matches expected process
+			continue NextProcess
 		}
 
 		if process[0] == '(' {
```

Comment:
can this still happen with `ss` ?

Reply:
hmm, I actually wasn't able to figure why ignore a listener process if its process name is (presumably) wrapped in parentheses. Couldn't really find much documentation on it.

Reply:
If you remove it does the test fail? If not maybe let's just rip it out and add it back if we figure out why it is needed :smile:

Reply:
:D sounds good. The test does not fail when I remove it.